fuel-vm/src/constraints/reg_key.rs

```diff
@@ -19,6 +25,35 @@ pub struct RegMut<'r, const INDEX: u8>(&'r mut Word);
 /// Immutable reference to a register value at a given index.
 pub struct Reg<'r, const INDEX: u8>(&'r Word);
 
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+/// A key to a writable register that is within
+/// the bounds of the writable registers.
+pub struct WriteRegKey(usize);
+
+impl WriteRegKey {
+    /// Create a new writable register key if the index is within the bounds
+    /// of the writable registers.
+    pub fn new(k: impl Into<usize>) -> Result<Self, RuntimeError> {
+        let k = k.into();
+        is_register_writable(&k)?;
+        Ok(Self(k))
+    }
+
+    fn translate(self) -> usize {
+        self.0 - VM_REGISTER_READ_COUNT
+    }
```

Comment:
It's because while there are 64 registers in total, the read registers are tracked in a separate array. It seems like this translate method converts from a global register index to a local register index within the set of writable registers.